inference_levels/InputEncoders/ESIM_encoder.py

```diff
@@ -0,0 +1,18 @@
+import torch
+from .input_encoding import BaseInputEncoder
+
+class ESIMInputEncoder(BaseInputEncoder):
```

Comment:
Besides the issue of being a duplication of BaseInputEncoder (without adding too much significant behavior), this class is not necessary:
logically, all it does is have some encoding_layer, which is applied twice, and that's it.
Instead of creating a class for this (which is more OOP but increases cognitive load, consider extracting this behaviour to your model class

`
def __init__(...):
self.encoding_layer = ... # LSTM is passed from outside

def __forward__(...):
  a_hat = self.encoding_layer(premise, h0, c0)
 b_hat = self.encoding_layer(hypothesis, h0, c0)

`

It is true that generally, we'd want to have a nice class/module for each separate entity, such as an "encoding step", however -- when the resulting classes are very shallow (have a small implementation and a large interface), then it is not worth it. You can read more about it In Chapter 4 of 'a Philosophy of Software Design'